### PR TITLE
feat: upgrade otel to v1.13

### PR DESCRIPTION
**Pull request by denysvitali**
### What did this pull request do?

Due to the breaking changes of OpenTelemetry, we need to upgrade this package to keep using it with newer versions of the SDK.

### PR Status

The PR wasn't tested (especially the metric part), but it compiles so 🤷‍♂️.
This might break stuff (there are no tests for the metrics part), but the tracing tests are still passing.

### CODE REVIEW

**Comment by zstone12** [comment]
@denysvitali  thank you very much for your PR ! :D
The CI check did not pass and you need to fix it.
lint-and-ut need Resubmit locally after formatting according to gofumpt.
`gofumpt -l -extra -w .`
It looks like we have to refine the unit tests in the metric part, is this something you want to do?

File: provider/provider.go [hunk 1]
```diff
@@ -6,16 +6,13 @@ import (
...
-	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
 	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
 	"go.opentelemetry.io/otel/metric/global"
-	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
-	"go.opentelemetry.io/otel/sdk/metric/export/aggregation"
-	processor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
-	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
```
Comment (zstone12) [inline provider/provider.go:13]: image
The reference to controller could not be found. how to make it  work?

Comment (denysvitali) [reply provider/provider.go:13]: Solved.

This is the result of running the example app now:

```
$ curl -s http://localhost:8088/metrics | grep '^go_sql'
go_sql_connections_closed_max_idle_time_total{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
go_sql_connections_closed_max_idle_total{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
go_sql_connections_closed_max_lifetime_total{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
go_sql_connections_idle{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 1
go_sql_connections_in_use{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
go_sql_connections_max_open{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
go_sql_connections_open{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 1
go_sql_connections_wait_count_total{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
go_sql_connections_wait_duration_total{otel_scope_name="opentelemetry/otel",otel_scope_version=""} 0
```

File: examples/metric/metrics/main.go [hunk 2]
```diff
@@ -105,24 +103,18 @@ func configureOpentelemetry() {
...
-				histogram.WithExplicitBoundaries(config.DefaultHistogramBoundaries),
-			),
-			aggregation.CumulativeTemporalitySelector(),
-			processor.WithMemory(true),
-		),
-	)
-
-	exporter, err := prometheus.New(config, ctrl)
+	exporter, err := prometheus.New()
+	if err != nil {
```
Comment (zstone12) [inline examples/metric/metrics/main.go:107]: Repeated error handling

File: provider/provider.go [hunk 3]
```diff
@@ -124,34 +118,25 @@ func NewOpenTelemetryProvider(opts ...Option) *otelProvider {
...
 		// metrics pusher
-		pusher := controller.New(
-			processor.NewFactory(
-				simple.NewWithHistogramDistribution(),
-				metricExp,
-			),
-			controller.WithResource(res),
-			controller.WithExporter(metricExp),
+		meterProvider := sdkmetric.NewMeterProvider(
+			sdkmetric.WithReader(sdkmetric.NewPeriodicReader(metricExp)),
```
Comment (zstone12) [inline provider/provider.go:129]: It makes the config variable `metricsClientOpts` unused. why not add `otlpmetricgrpc.WithAggregationSelector(sdkmetric.DefaultAggregationSelector)` to `metricsClientOpts`

Comment (denysvitali) [reply provider/provider.go:131]: Done, thanks!

**Comment by zstone12** [comment]
LGTM!
Thank you again for the PR!
The unit tests in the metric part will probably be added later:D

**Pull request merged**